devito/operations/interpolators.py

```diff
@@ -126,90 +122,96 @@ def __init__(self, sfunction):
     def grid(self):
         return self.sfunction.grid
 
-    @cached_property
+    @property
+    def _weights(self):
+        raise NotImplementedError
+
+    @property
+    def _psym(self):
+        return self.sfunction._point_symbols
+
+    @property
+    def _gdim(self):
+        return self.grid.dimensions
+
+    def implicit_dims(self, implicit_dims):
+        return as_tuple(implicit_dims) + self.sfunction.dimensions
```

Comment:
you know I'm thinking, maybe instead of `sfunction` we could use just `function`, like we do everywhere else

Reply:
indeed we should change that